### ISSUE TITLE
Add a hook after loading but before adding to diffsync store for `NautobotAdapter`

### ISSUE THREAD

**Issue by Kircheneer** [nautobot/nautobot-app-ssot#399]
### Use Case

For example injecting diffsync flags like `SKIP_UNMATCHED_DST` on an object-level.

**Comment by jdrew82** [comment]
How is this different from DiffSyncModelFlags?

**Comment by itdependsnetworks** [comment]
The linked code is added to all instances (I think). I kinda remember thinking the same thing as Leo, but it's been a while. I had it worked out in my head a year or two ago, but I think it is likely to be a pattern of some sort. Similar to how permissions or dynamic groups are, some was to map these flags, so they get applied based on the data of the instance.

**Comment by Kircheneer** [comment]
Yes, it is already technically possible to do this on a per instance basis, even with contrib. However, the goal with contrib was to eliminate the possibility to do things your own way and provide opinionated defaults. The intent of this issue was to provide such a default for this specific, semi-common use case. In the different integrations it is done in a non-standardized manner, and the queries used may be suboptimal, posing potential n+1 scenarios, which I wanted to eliminate with contrib, as they inflate the load time of the Nautobot adapter.
